src/AppInstallerCLICore/Argument.cpp

```diff
@@ -216,8 +216,10 @@ namespace AppInstaller::CLI
             return { type, "dependency-source"_liv, ArgTypeCategory::Source };
         case Execution::Args::Type::CustomHeader:
             return { type, "header"_liv, ArgTypeCategory::Source };
+
+        // This must specifically excluded from being a source argument due to dependencies in local manifests potentially requiring a source search and source agreements

```

Comment:
Does this mean this dependency-source (line 216) would also fail? dependency-source is specifically created for local manifest with dependencies testing purpose.

Anyway, this looks like a hacky fix, we should probably fix in the actual validation code instead of removing a correctly assigned tag

Reply:
I guess the quesion then is whether the tag is qctually correctly assigned, as it caused a semi-breaking behavior (according to the linked issue). 

If we are fine with leaving these as being mutually exclusive arguments, then this PR and the linked issue should be closed. 

If we need to fix the regression, then I'm not sure the best way to do it if this isn’t the desired path forward

Reply:
`winget install -m <manifest> -s <winget>` is not valid, but `winget install -m <manifest> --dependency-source <winget>` is valid. I think we need to assign more granular tags to fix the issue correctly.

Reply:
But then what about when the dependency source requires agreements?

Reply:
--accept-source-agreements should be allowed too

Reply:
Ah, now I see what you mean about needing to be more granular. 

Because of the `ArgTypeCategory::Source` both `--dependency-source` and `--accept-source-agreements` are unable to be used with the `--manifest` argument, but they are both correctly tagged with `ArgTypeCategory::Source` as they relate to the behavior of sources. Is that accurate? If it is accurate, I would still believe that `--accept-source-agreements` shouldn’t be tagged as a source argument because 1) It isn’t selecting a source and 2) it isn’t providing any data like a header, it's simply a flag stating how the CLI should behave *if* a source agreement is required. I'm thinking of it like `--force` where just because it is added doesn’t mean it will do anything - if there is nothing that needs to be forced then it is effectively ignored; if there are no source agreements - the behavior is the same whether the argument is present or not.

-----

I do agree that the dependency source shpuld be able to be used with local manifests, and based on the current implementation it cannot. 

So in order to be more granular we would either have to add an ArgTypeCategory for indicating that it should still be allowed with `--manifest` (or do the opposite and make allowed the default, flagging the ones that are incompatible). 

Or I suppose we could add a new argtypecategory that better describes what each does - but then I would think that there would need to be a specific one for agreements, and one for dependency source, since those source properties are allowed with local manifests while others aren't - in which case, being their own categories, it would be more straightforward to simply leave them untagged.

Reply:
@yao-msft - Any recommendations on how the additional granularity should be implemented? I ran into the issue with `--dependency-source` not working with `--manifest` today, so I'd like to get the issue resolved

<!--
cc @denelon 
-->

Reply:
From thinking about it for, like, five minutes..
I think I would change the `ArgTypeCategory`.

Replace `ArgTypeCategory::Source` with
* `ArgTypeCategory::SourceForQuery` - for args that refer specifically to the source we're initially querying
* `ArgTypeCategory::SourceOrDependencySource` - for args that apply to either the initial source or the dependencies

Then we can have the validation be
```c++
        if (WI_IsFlagSet(categories, ArgTypeCategory::Manifest) &&
            WI_IsAnyFlagSet(categories, ArgTypeCategory::PackageQuery | ArgTypeCategory::SourceForQuery)) 
```

Then tag
`--source, --header? = SourceForQuery`
`--dependency-source, --accept-source-agreements = SourceOrDependencySource`

I'm not sure if we'd need to do any validation on `SourceOrDependencySource`

Reply:
Ah, that makes perfect sense.